lesson11/main.js

```diff
@@ -15,61 +15,48 @@ const loaded = () => {
   loader.classList.add("loaded");
 };
 
-window.onload = () => {
-  loading();
-
-  const fetchedData = new Promise((resolve, reject) => {
-    setTimeout(() => {
-      resolve(listItems());
-    }, 2500);
+const jsonUrl = "https://jsondata.okiba.me/v1/json/PBH3M210806155341";
+
+const createNewList = (data) => {
+  const ul = document.getElementById("js-lists");
+  const fragment = document.createDocumentFragment();
+
+  data.forEach((key) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+
+    a.href = `/${key.a}`;
+    a.textContent = key.text;
+    img.src = key.img;
+    img.alt = key.alt;
+
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
   });
+  ul.appendChild(fragment);
+};
 
-  const listItems = async () => {
-    const jsonUrl = "https://jsondata.okiba.me/v1/json/PBH3M210806155341";
+const fetchedData = async () => {
+  try {
+    loading();
     const response = await fetch(jsonUrl);
     const json = await response.json();
     return json.data;
```

Comment:
IMO

I think it is safe to assume that try-catch encloses processes that may cause errors or exceptions in a try block, and when an error occurs in the try block, the process is transferred to catch.

Therefore, it may be more natural to cut out loading, instead of including it in the try.

Reply:
It might be better to run load outside of try. 
Sorry about that.

Reply:
@shouyamamoto
Thanks again.I removed it from try part!

@riku9228
No worries! I appreciated :)

[5a4a13c](https://github.com/yuka830/js-lessons/pull/13/commits/5a4a13cb2f801eacc8250eb8f6a272ae16efd660)
[CodeSandbox](https://codesandbox.io/s/lesson11-d7w3u?file=/main.js:0-1673)

Reply:
IMO

```js
const fetchedData = async () => {
  loading();　<-- this
  try {
    const response = await fetch(jsonUrl);
    const json = await response.json();
    return json.data;
  } catch (error) {
    const wrapper = document.getElementById("js-wrapper");
    wrapper.textContent = "データの取得ができませんでした。";
    console.error(error);
  } finally {
    loaded();
  }
};
```

This is just my idea, so you don't have to modify it, but if you are convinced, please refer to it.

When init() is executed, it goes to get the data, right?
That state is represented by loading(), so I think it is more natural to execute loading in init.

I think it's more natural to execute loading in init.
"I wanted to say, "You don't have to include it in the try.

Reply:
You're right! It's more natural way, so I've changed!
I wanna be able to think the way you do...thanks a lot :)
[87cb01e](https://github.com/yuka830/js-lessons/pull/13/commits/87cb01ef0f677dd8cbdfb4c9192fa6637c656df9)